apps/web/src/components/pure/workspace-slider-bar/index.tsx

```diff
@@ -64,117 +73,162 @@ export const WorkSpaceSliderBar: React.FC<WorkSpaceSliderBarProps> = ({
 }) => {
   const currentWorkspaceId = currentWorkspace?.id || null;
   const { t } = useTranslation();
-  const [sidebarOpen] = useSidebarStatus();
+  const [sidebarOpen, setSidebarOpen] = useSidebarStatus();
   const pageMeta = usePageMeta(currentWorkspace?.blockSuiteWorkspace ?? null);
   const onClickNewPage = useCallback(async () => {
     const page = await createPage();
     openPage(page.id);
   }, [createPage, openPage]);
+  const theme = useTheme();
+  const floatingSlider = useMediaQuery(theme.breakpoints.down('md'));
+  const [sliderWidth, setSliderWidth] = useSidebarWidth();
+  const [isResizing, setIsResizing] = useSidebarResizing();
+  const show = isPublicWorkspace ? false : sidebarOpen;
+  const actualWidth = show
+    ? floatingSlider
+      ? 'calc(10vw + 400px)'
+      : sliderWidth
+    : 0;
   return (
     <>
-      <StyledSliderBar show={isPublicWorkspace ? false : sidebarOpen}>
-        <StyledSidebarSwitchWrapper>
-          <SidebarSwitch
-            visible={sidebarOpen}
-            tooltipContent={t('Collapse sidebar')}
-            testid="sliderBar-arrowButton-collapse"
-          />
-        </StyledSidebarSwitchWrapper>
+      <StyledSliderBarWrapper>
+        <StyledSliderBar
+          resizing={isResizing}
+          floating={floatingSlider}
+          style={{ width: actualWidth }}
```

Comment:
can we use `sx`?

Reply:
I tried `sx`as well as passing width as a parameter to styled-component. Looks like it will create a new class for every new width. I guess it is better to use inline-styles in this case.

https://user-images.githubusercontent.com/584378/228752960-c26fd546-7152-4f34-a5a8-e5220a3512b0.mp4